### PR TITLE
New package: matterhorn-50200.7.0

### PR DESCRIPTION
**Pull request by stacyharper**
related: https://github.com/void-linux/void-packages/issues/20115

### CODE REVIEW

**Comment by Eluminae** [comment]
log: https://dav.misterbanal.net/pub/d7432b44-3927-4ab0-95a5-8ee172d011c4.gkXkXH6Nto.txt

**Comment by Eluminae** [comment]
@ametisf 

This is not stack packaged

I got: 

```
Error parsing targets: The specified targets matched no packages.
Perhaps you need to run 'stack init'?
=> ERROR: matterhorn-50200.7.0_1: do_install: 'STACK_ROOT=$wrksrc/.stack stack ${_stack_args} install ${make_build_args} --local-bin-path=${DESTDIR}/usr/bin' exited with 1
=> ERROR:   in do_install() at common/build-style/haskell-stack.sh:35
```

Is stack cabal compatible ?

**Comment by ametisf** [comment]
I don't see the whole error in your last comment so I can't tell exactly but, Stack doesn't have to have the package you're trying to build it has to have all it's dependencies and it's not the case for matterhorn anyway so that won't work.

The error you get with cabal is caused by it being built as a dynamically linked binary, all the haskell libraries are being linked dynamicaly, and then you don't include them in the package, xbps goest to search for them and then it tells you it couldn't find any package providing those libraries. The solution is (since those libraries wouldn't be useful to anything else) is to link the binary statically.

This made it build properly
```
diff --git a/srcpkgs/matterhorn/template b/srcpkgs/matterhorn/template
index 83dbcd0aac..95681b649c 100644
--- a/srcpkgs/matterhorn/template
+++ b/srcpkgs/matterhorn/template
@@ -16,7 +16,7 @@ nopie_files="/usr/bin/matterhorn"
 do_build() {
        rm -f cabal.project
        cabal new-update
-       cabal new-build -j --enable-shared --enable-executable-dynamic --ghc-options=-dynamic
+       cabal new-build --jobs="${XBPS_MAKEJOBS}" --disable-executable-dynamic
 }

 do_install() {
```
but there are more things that should be fixed about your template, definitely installing with `vbin "$(find ...)"` is a bad idea, either change it to the exact location of the output binary or better use the `cabal new-install` subcommand if it's possible to give it a `$DESTDIR`.

**Comment by Eluminae** [comment]
> I don't see the whole error in your last comment so I can't tell exactly but, Stack doesn't have to have the package you're trying to build it has to have all it's dependencies and it's not the case for matterhorn anyway so that won't work.

Yeah I just tried to play with the lts stack variable without
understanding what I was doing, stucked on the "omg it miss all of those
dependencies" step.

> The error you get with cabal is caused by it being built as a dynamically linked binary, all the haskell libraries are being linked dynamicaly, and then you don't include them in the package, xbps goest to search for them and then it tells you it couldn't find any package providing those libraries. The solution is (since those libraries wouldn't be useful to anything else) is to link the binary statically.
> 
> This made it build properly
> ```
> diff --git a/srcpkgs/matterhorn/template b/srcpkgs/matterhorn/template
> index 83dbcd0aac..95681b649c 100644
> --- a/srcpkgs/matterhorn/template
> +++ b/srcpkgs/matterhorn/template
> @@ -16,7 +16,7 @@ nopie_files="/usr/bin/matterhorn"
>  do_build() {
>         rm -f cabal.project
>         cabal new-update
> -       cabal new-build -j --enable-shared --enable-executable-dynamic --ghc-options=-dynamic
> +       cabal new-build --jobs="${XBPS_MAKEJOBS}" --disable-executable-dynamic
>  }
> 
>  do_install() {
> ```

Oh yeah, it feels always better when you understand what your doing.
Thanks a lot for you help, you made it working!

> but there are more things that should be fixed about your template, definitely installing with `vbin "$(find ...)"` is a bad idea, either change it to the exact location of the output binary or better use the `cabal new-install` subcommand if it's possible to give it a `$DESTDIR`.

I agree, I'll try something better. It seems there is a destdir cabal
argument but I think using the bin path a more minimalist way (but the
full path is horrible x))

Thanks again for your help!

File: srcpkgs/matterhorn/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'matterhorn'
+pkgname=matterhorn
+version=50200.7.0
+revision=1
+makedepends="cabal-install ghc zlib-devel ncurses-devel"
+depends="gmp zlib"
```
Comment (ametisf) [inline srcpkgs/matterhorn/template]: You already depend on zlib in makedepends, gmp should be picked up automatically too if something links against it.

File: srcpkgs/matterhorn/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+nopie_files="/usr/bin/matterhorn"
+
+do_build() {
+	rm -f cabal.project
+	cabal new-update
+	cabal new-build --jobs="${XBPS_MAKEJOBS}" --disable-executable-dynamic
+}
+
+do_install() {
+	vbin dist-newstyle/build/x86_64-linux/ghc-8.6.5/matterhorn-50200.7.0/x/matterhorn/build/matterhorn/matterhorn
```
Comment (ametisf) [inline srcpkgs/matterhorn/template]: This won't work, xbps supports more architectures than just x86_64. I'm sure cabal has better support for installing binaries than that.

File: srcpkgs/matterhorn/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'matterhorn'
+pkgname=matterhorn
+version=50200.7.0
+revision=1
+makedepends="cabal-install ghc zlib-devel ncurses-devel"
+depends="gmp zlib"
+short_desc="A terminal-based chat client for MatterMost"
```
Comment (ametisf) [inline srcpkgs/matterhorn/template]: Use xlint from xtools.

Comment (Eluminae) [reply srcpkgs/matterhorn/template]: Okay this will save my future

File: srcpkgs/matterhorn/template (hunk 1 above)
Comment (Eluminae) [reply srcpkgs/matterhorn/template]: True, I removed it

File: srcpkgs/matterhorn/template (hunk 2 above)
Comment (Eluminae) [reply srcpkgs/matterhorn/template]: I'm trying to use cabal with some sort of distdir or bindir. I leave it this way for tonight. I'll try again tomorrow, reading seriously documentations.

**Comment by abenson** [comment]
Any movement on getting this fixed? ALso 50200.8.0 is out

**Comment by Eluminae** [comment]
I just can't find a way to fix the bin path. It is relative to the infra plus the ghc version and as it seems I'm not allowed to use wildcard so I can't fix this with my low ghc knowledges.

If anyone want to try and find another way to do this.

**Pull request closed without merging**